module/scmi_reset_domain/src/mod_scmi_reset_domain.c

```diff
@@ -0,0 +1,630 @@
+
+    outmsg.latency = reset_dev_config->latency;
+
+    strncpy((char *)outmsg.name, fwk_module_get_name(reset_device->element_id),
+            sizeof(outmsg.name) - 1);
+
+    outmsg.status = SCMI_SUCCESS;
+    outmsg_size = sizeof(outmsg);
+
+exit:
+    scmi_rd_ctx.scmi_api->respond(service_id, &outmsg, outmsg_size);
+
+    return status;
+}
+
+static int reset_request_handler(fwk_id_t service_id,
+                                 const uint32_t *payload)
+{
+    int status;
+    unsigned int agent_id;
+    struct mod_reset_domain_dev_config *reset_dev_config;
+    struct scmi_reset_domain_request_a2p params = { 0 };
+    struct scmi_reset_domain_request_p2a outmsg = {
+        .status = SCMI_NOT_FOUND
+    };
+    enum mod_reset_domain_mode mode = MOD_RESET_DOMAIN_MODE_EXPLICIT_DEASSERT;
+
+    size_t outmsg_size = sizeof(outmsg.status);
+    const struct mod_reset_domain_api *reset_api = scmi_rd_ctx.reset_api;
+    const struct mod_scmi_reset_domain_device *reset_device = NULL;
+
+    params = *(const struct scmi_reset_domain_request_a2p *)payload;
+
+    status = scmi_rd_ctx.scmi_api->get_agent_id(service_id, &agent_id);
+    if (status != FWK_SUCCESS)
+        goto exit;
+
+    if (params.domain_id >= scmi_rd_ctx.plat_reset_domain_count) {
+        status = FWK_E_PARAM;
+        goto exit;
+    }
+
+    status = get_reset_device(service_id, params.domain_id, &reset_device);
+    if (status != FWK_SUCCESS)
+        goto exit;
+
+    reset_dev_config = (struct mod_reset_domain_dev_config *)
+                       fwk_module_get_data(reset_device->element_id);
+
+    /* Check if explicit assert is requested.
+     * Note, valid request in flags
+     *       b000 Explicit de-assert request.
+     *       b001 Auto Reset request.
+     *       b010 Explicit reset request.
+     *       b101 Auto Reset Async.
+     */
+    if (!(params.flags & SCMI_RESET_DOMAIN_AUTO)) {
+        /* If auto reset is not requested then check if device supports explicit
+         * assert/de-assert reset.
+         */
+        if (!(reset_dev_config->modes &
+            (MOD_RESET_DOMAIN_MODE_EXPLICIT_ASSERT |
+            MOD_RESET_DOMAIN_MODE_EXPLICIT_DEASSERT))) {
+
+            outmsg.status = SCMI_NOT_SUPPORTED;
+            goto exit;
+        } else {
+           if (params.flags & SCMI_RESET_DOMAIN_EXPLICIT)
+               mode = MOD_RESET_DOMAIN_MODE_EXPLICIT_ASSERT;
+        }
+    } else {
+        mode = SCMI_RESET_DOMAIN_AUTO;
+    }
+
+    /* Handle async reset request. */
+    if (params.flags & SCMI_RESET_DOMAIN_ASYNC) {
+        /* Async reset request is valid only in Autoreset mode
+         */
+        if (!(params.flags & SCMI_RESET_DOMAIN_AUTO)) {
+            outmsg.status = SCMI_INVALID_PARAMETERS;
+            goto exit;
+        }
+
+        /* Return not supported as associated device does not support
+         * Async reset.
+         */
+        if (!(reset_dev_config->modes &
+            MOD_RESET_DOMAIN_MODE_AUTO_RESET_ASYNC)) {
```

Comment:
Thanks for this, However indentation is correct because the line break is for expression and the next line should start below the start of expression, so not changing